src/core/Modal/Modal/Modal.tsx

```diff
@@ -1,24 +1,23 @@
-import React, { Component, ReactNode } from 'react';
-import ReactDOM from 'react-dom';
+import React, { Component, ReactNode, createRef } from 'react';
 import { default as styled } from 'styled-components';
-import { createFocusTrap, FocusTrap } from 'focus-trap';
+import { default as ReactModal } from 'react-modal';
 import classnames from 'classnames';
-import { HtmlDiv, HtmlDivProps, HtmlDivWithRef } from '../../../reset';
-import { AutoId } from '../../../utils/AutoId';
-import { windowAvailable } from '../../../utils/common';
 import { ModalContent, ModalFooter } from '../';
 /* Styles for disabling scrolling. See file for more info. */
 import './Modal.globals.scss';
 import { baseStyles } from './Modal.baseStyles';
 
 export type ModalVariant = 'smallScreen' | 'default';
 
-export interface ModalProps
-  extends Omit<HtmlDivProps, 'children' | 'className' | 'title' | 'ref'> {
+export interface ModalProps {
   /** Show or hide the modal */
   visible: boolean;
+  /** Application root element id for setting aria-hidden when necessary */
+  appElementId?: string;
```

Comment:
Should this prop be mandatory? Implementation breaks if it's empty, but there is a test before passing it to the underlying react-modal. Might make testing more difficult if this is changed to mandatory.

Reply:
I would vote for it to be mandatory, as we should guide the usage of the component with the props.
It's unfortunate if testing will then come more difficult, but hopefully there is another way to have it tested.

Reply:
Our tests should mostly already support this, I was referring to testing when using this component in projects.

Reply:
This is now mandatory prop. However, there is still a truthy check before applying it, because things break silently and more badly if appElement id has a bad value.

Reply:
And now there is also a runtime check for the prop and a logger.error if the prop is not truthy.